src/test/java/org/jenkinsci/plugins/durabletask/WindowsBatchScriptTest.java

```diff
@@ -98,4 +101,69 @@ private void testWithPath(String path) throws Exception {
         c.cleanup(ws);
     }
 
-}
\ No newline at end of file
+    @Issue("JENKINS-34150")
+    @Test(timeout = 5000) public void batchScriptHanging() throws Exception {
+        final int concurrency = 4;
+
+        StreamTaskListener listener = StreamTaskListener.fromStdout();
+        FilePath ws = j.jenkins.getRootPath().child("ws");
+        Launcher launcher = j.jenkins.createLauncher(listener);
+        WindowsBatchScript script = new WindowsBatchScript("ping 127.0.0.1 -n 2");
+
+        List<Controller> controllers = Arrays.asList(new Controller[concurrency]);
+        EnvVars envVars = new EnvVars();
+        for (int i = 0; i < concurrency; i++) {
+            Controller controller = script.launch(envVars, ws, launcher, listener);
+            controller.writeLog(ws, System.out);
+            controllers.set(i, controller);
+        }
+
+        boolean stillRunning = true;
+        while (stillRunning) {
+            stillRunning = false;
+            for (int i = 0; i < concurrency; i++) {
+                Controller controller = controllers.get(i);
+                boolean running = (controller.exitStatus(ws, launcher) == null);
+                stillRunning |= running;
+            }
+        }
+
+        for (int i = 0; i < concurrency; i++) {
+            controllers.get(i).cleanup(ws);
+        }
+    }
+
+    /**
+     * This issue came up while trying to fix JENKINS-34150. The problem was that the termination of the batch file
+     * did not work properly anymore. This is the reason why it is referencing to the same issue.
+     *
+     * @throws Exception
+     */
+    @Issue("JENKINS-34150")
+    @Test(timeout=5000) public void batchScriptForcedTerminate() throws Exception {
+        StreamTaskListener listener = StreamTaskListener.fromStdout();
+        FilePath ws = j.jenkins.getRootPath().child("ws");
+        Launcher launcher = j.jenkins.createLauncher(listener);
+        WindowsBatchScript script = new WindowsBatchScript("TIMEOUT /T 10");
```

Comment:
Right. This was not a issue, because the scenario this test was working in spawned a new window that worked as input context. The command can easily be changed to a `ping` or something like this to work around that. I just used the timeout to get a defined runtime.